wrangler-core/src/main/java/io/cdap/wrangler/executor/RecipePipelineExecutor.java

```diff
@@ -161,4 +186,38 @@ private List<Directive> getDirectives() throws RecipeException {
     }
     return directives;
   }
+
+  private Schema generateOutputSchema(Schema inputSchema, List<Row> output) throws RecordConvertorException {
+    Set<Field> outputFields = new HashSet<>();
+    for (Row row : output) {
+      for (Pair<String, Object> rowField : row.getFields()) {
+        String fieldName = rowField.getFirst();
+        Object fieldValue = rowField.getSecond();
+
+        Schema existing = inputSchema.getField(fieldName) != null ? inputSchema.getField(fieldName).getSchema() : null;
+        Schema generated = fieldValue != null && (existing == null || !isValidSchemaForValue(existing, fieldValue)) ?
+          generator.getSchema(fieldValue, fieldName) : null;
```

Comment:
You may want to change existing to nullable if it is not nullable and current value is null

Reply:
I believe this is still applicable what should happen if all values are null, but the original schema is non-nullable